pkg/controller/statusmanager/status_manager.go

```diff
@@ -63,6 +67,39 @@ func (status *StatusManager) set(reachedAvailableLevel bool, conditions ...confi
 			return err
 		}
 
+		// Check for differences between current and rendered related objects, delete if necessary
+		var found bool = false
+		for _, current_obj := range co.Status.RelatedObjects {
+			for _, rendered_obj := range status.relatedObjects {
+				found = reflect.DeepEqual(current_obj, rendered_obj)
+
+				if found {
+					break
+				}
+			}
+			if !found {
+				log.Printf("Detected related object not rendered by manifests, deleting...")
+				gvr := schema.GroupVersionResource{
+					Group:    current_obj.Group,
+					Resource: current_obj.Resource,
+				}
+				gvk, err := status.mapper.KindFor(gvr)
+				if err != nil {
+					log.Printf("Error getting GVK of object for deletion: %v", err)
+					continue
+				}
+				obj_to_delete := &uns.Unstructured{}
+				obj_to_delete.SetName(current_obj.Name)
+				obj_to_delete.SetNamespace(current_obj.Namespace)
+				obj_to_delete.SetGroupVersionKind(gvk)
+				err = status.client.Delete(context.Background(), obj_to_delete)
```

Comment:
This still doesn't do background deletes. They are a Delete option - check the signature of status.client.Delete. The `context.Background()` here is incorrect - it should be `context.TODO`.